presto-main/src/main/java/io/prestosql/execution/scheduler/SourcePartitionedScheduler.java

```diff
@@ -317,7 +317,7 @@ else if (pendingSplits.isEmpty()) {
         //     which may contain recently published splits. We must not ignore those.
         //   * If any scheduleGroup is still in DISCOVERING_SPLITS state, it means it hasn't realized that there will be no more splits.
         //     Next time it invokes getNextBatch, it will realize that. However, the invocation will fail we tear down splitSource now.
-        if ((state == State.NO_MORE_SPLITS || state == State.FINISHED) || (noMoreScheduleGroups && scheduleGroups.isEmpty() && splitSource.isFinished())) {
+        if ((state == State.NO_MORE_SPLITS || state == State.FINISHED) || (noMoreScheduleGroups && scheduleGroups.isEmpty())) {
```

Comment:
@dain does this look correct to you?

It seems that `SplitSource#isFinished()` is redundant as split scheduler should drain splits from all partitions/buckets anyway.